cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -128,10 +131,14 @@ public AdapterTypeInfo getAdapterTypeInfo(String adapterType) {
    * @param namespace namespace to lookup the adapter
    * @param adapterName name of the adapter
    * @return requested {@link AdapterSpecification} or null if no such AdapterInfo exists
+   * @throws AdapterNotFoundException if the requested adapter is not found
    */
-  @Nullable
-  public AdapterSpecification getAdapter(String namespace, String adapterName) {
-    return store.getAdapter(Id.Namespace.from(namespace), adapterName);
+  public AdapterSpecification getAdapter(String namespace, String adapterName) throws AdapterNotFoundException {
+    AdapterSpecification adapterSpecification = store.getAdapter(Id.Namespace.from(namespace), adapterName);
+    if (adapterSpecification == null) {
+      throw new AdapterNotFoundException(String.format("Adapter %s not found.", adapterName));
```

Comment:
how come we want to throw an exception instead of just returning null?  Is that what we do for other types of specs?